runtime/near-vm-runner/src/cache.rs

```diff
@@ -228,9 +254,29 @@ pub mod wasmer2_cache {
         code: &[u8],
         config: &VMConfig,
         store: &wasmer::Store,
-    ) -> Result<wasmer::Module, VMError> {
-        let prepared_code = prepare::prepare_contract(code, config)?;
-        wasmer::Module::new(&store, prepared_code).map_err(|err| err.into_vm_error())
+    ) -> Result<wasmer::Module, CompilationError> {
+        let prepared_code =
+            prepare::prepare_contract(code, config).map_err(CompilationError::PrepareError)?;
+        wasmer::Module::new(&store, prepared_code).map_err(|err| match err {
+            wasmer::CompileError::Wasm(_) => {
+                CompilationError::WasmerCompileError { msg: err.to_string() }
+            }
+            wasmer::CompileError::Codegen(_) => {
+                CompilationError::WasmerCompileError { msg: err.to_string() }
+            }
+            wasmer::CompileError::Validate(_) => {
+                CompilationError::WasmerCompileError { msg: err.to_string() }
+            }
+            wasmer::CompileError::UnsupportedFeature(_) => {
+                CompilationError::WasmerCompileError { msg: err.to_string() }
+            }
+            wasmer::CompileError::UnsupportedTarget(_) => {
+                CompilationError::WasmerCompileError { msg: err.to_string() }
+            }
+            wasmer::CompileError::Resource(_) => {
```

Comment:
@olonho this preserves the existing behavior, but just makes it explicit what's going on. And, looking at this, it seems we might not actually treat some of these variants as deterministic errors?

Reply:
Checked that, it seems this is mostly deterministic error, although can happen in memory limited situations with resource constraints in some OSes (i.e. Linux). Guess we could keep it as it.